### PR TITLE
doc:Add to block=storage and create alert-manager configuration docum…

### PR DESCRIPTION
**Pull request by mizunos**
…entation

**What this PR does**:
Added information about alertmanager and ruler using block storage and step to configure alertmanager using cortextool

File touched
docs/block-storage
docs/guides/alert-manager-configuration.md

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Tests updated
- [ x] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/guides/alert-manager-configuration.md
Comment (pstibrany) [inline docs/guides/alert-manager-configuration.md:16]: > The following describe the step ...

Suggestion: "The following example shows the steps to load ...".

> AlertManager

We should not capitalize "Manager". 

> Follow the instruction above ...

There was no instruction, only link to Cortex Tool.

File: docs/guides/alert-manager-configuration.md
Comment (pstibrany) [inline docs/guides/alert-manager-configuration.md:21]: This creates block quote. Perhaps we should use sub-headings instead?

File: docs/guides/alert-manager-configuration.md
Comment (pstibrany) [inline docs/guides/alert-manager-configuration.md:23]: Let's remove `>>>`, and make it a normal text instead of triple-blockquote. Same for the text below.

File: docs/guides/alert-manager-configuration.md
Comment (pstibrany) [inline docs/guides/alert-manager-configuration.md:77]: Let's simplify the link to `https://grafana.com/blog/2020/02/25/step-by-step-guide-to-setting-up-prometheus-alertmanager-with-slack-pagerduty-and-gmail` only.

File: docs/guides/alert-manager-configuration.md
Comment (pstibrany) [inline docs/guides/alert-manager-configuration.md:79]: Subheading.

File: docs/guides/alert-manager-configuration.md
Comment (pstibrany) [inline docs/guides/alert-manager-configuration.md:93]: Subheading.

**Review by pstibrany — COMMENTED** [review]
Thank you for working on documentation, much appreciated!

Let's please fix the markdown formatting in alert-manager-configuration.md (use of subheadings, remove blockquotes, end sentences with period) before proceeding.

**Comment by mizunos** [comment]
Made the changes as suggested.

File: docs/guides/alert-manager-configuration.md
Comment (pracucci) [inline docs/guides/alert-manager-configuration.md:5]: The slug should be all lowercase.

```suggestion
slug: alertmanager-configuration
```

File: docs/guides/alert-manager-configuration.md
Comment (pracucci) [inline docs/guides/alert-manager-configuration.md:2]: The alertmanager is spelled "Alertmanager" in Prometheus ([see here](https://prometheus.io/docs/alerting/latest/alertmanager/)) so I would change "AlertManger" to "Alertmanager" across this PR to keep it consistent with Prometheus.

File: docs/guides/alert-manager-configuration.md
Comment (pracucci) [inline docs/guides/alert-manager-configuration.md:14]: Links should be relative in the doc. Something like this:

```suggestion
Cortex AlertManager can be uploaded via [Cortex Set Alertmanager API](../api/_index.md#set-alertmanager-configuration) or using Grafana Labs [Cortex Tools](https://github.com/grafana/cortex-tools).
```

File: docs/guides/alert-manager-configuration.md
Comment (pracucci) [inline docs/guides/alert-manager-configuration.md:102]:
```suggestion
To upload the configuration for Cortex `alertmanager` using Cortex API and curl - see [Cortex Set Alertmanager API](../api/_index.md#set-alertmanager-configuration).
```

File: docs/guides/alert-manager-configuration.md
Comment (pracucci) [inline docs/guides/alert-manager-configuration.md:28]: I'm wondering if we should use a simpler alertmanager config as an example. The purpose of this guide is not to show how Alertmanager works, but how to configure it in Cortex. I've the feeling such a long example config may deviate the focus from what we want to show in this guide.

For example, the following in a SMTP-based receiver:

```
global:
  # The smarthost and SMTP sender used for mail notifications.
  smtp_smarthost: 'localhost:25'
  smtp_from: 'alertmanager@example.org'
  smtp_auth_username: 'alertmanager'
  smtp_auth_password: 'password'

route:
  # A default receiver.
  receiver: send-email

receivers:
  - name: send-email
    email_configs:
      - to: 'someone@localhost'
```

Comment (mizunos) [reply docs/guides/alert-manager-configuration.md:28]: Precisely my issue that motivates me to provide a more "real" example.  Lots of use cases are slack based and I spent countless hours trying to find a more "real" life setup to help with my application of Cortex.  So by providing a little more 'real" example,  I hope to help the users sometimes to comb through many many messages and issued to find the answer

File: docs/blocks-storage/_index.md
Comment (pstibrany) [reply docs/blocks-storage/_index.md:34]: Let's call it "object storage", not "block storage" -- we use that term for storing TSDB blocks.

File: docs/guides/alert-manager-configuration.md
Comment (pracucci) [reply docs/guides/alert-manager-configuration.md:28]: I understand this, but to me it looks more like a topic for improving the Prometheus alertmanager doc.

File: docs/blocks-storage/_index.md
Comment (mizunos) [reply docs/blocks-storage/_index.md:34]: Ok. Sounds good. Done!

File: docs/guides/alert-manager-configuration.md
Comment (mizunos) [reply docs/guides/alert-manager-configuration.md:5]: Done

File: docs/guides/alert-manager-configuration.md
Comment (mizunos) [reply docs/guides/alert-manager-configuration.md:102]: Done

File: docs/guides/alert-manager-configuration.md
Comment (mizunos) [reply docs/guides/alert-manager-configuration.md:28]: That's fine by me.
I got my answer :-)

**Comment by mizunos** [comment]
@pracucci @pstibrany - anybody review this yet?

File: docs/guides/alert-manager-configuration.md
Comment (pracucci) [inline docs/guides/alert-manager-configuration.md:14]:
```suggestion
Cortex Alertmanager can be uploaded via Cortex [Set Alertmanager configuration API](../api/_index.md#set-alertmanager-configuration) or using Grafana Labs [Cortex Tools](https://github.com/grafana/cortex-tools).
```

File: docs/guides/alert-manager-configuration.md
Comment (pracucci) [inline docs/guides/alert-manager-configuration.md:8]:
```suggestion
## Context
```

File: docs/guides/alert-manager-configuration.md
Comment (pracucci) [inline docs/guides/alert-manager-configuration.md:12]:
```suggestion
### Cortex Alertmanager configuration
```

File: docs/guides/alert-manager-configuration.md
Comment (pracucci) [inline docs/guides/alert-manager-configuration.md:23]:
```suggestion
#### 1. Create the Alertmanager configuration `.yml` file
```

File: docs/guides/alert-manager-configuration.md
Comment (pracucci) [inline docs/guides/alert-manager-configuration.md:47]:
```suggestion
#### 2. Upload the Alertmanager configuration
```

File: docs/guides/alert-manager-configuration.md
Comment (pracucci) [inline docs/guides/alert-manager-configuration.md:61]:
```suggestion
#### 3. Ensure the configuration has been uploaded correctly
```

File: docs/guides/alert-manager-configuration.md
Comment (pracucci) [inline docs/guides/alert-manager-configuration.md:59]:
```suggestion
If there is no error reported, the upload is successful.
```

File: docs/guides/alert-manager-configuration.md
Comment (pracucci) [inline docs/guides/alert-manager-configuration.md:70]: This should be moved to the step (2) above.

**Review by pracucci — COMMENTED** [review]
Thanks a lot for addressing my feedback 🙏  I left some final nits and then we're good to me. Please fix the DCO commit signing (see https://cortexmetrics.io/docs/contributing/#developer-certificates-of-origin-dcos), otherwise we'll not be able to merge it.

File: docs/guides/alert-manager-configuration.md
Comment (mizunos) [reply docs/guides/alert-manager-configuration.md:14]: Done

File: docs/guides/alert-manager-configuration.md
Comment (mizunos) [reply docs/guides/alert-manager-configuration.md:59]: @pracucci - you really pay attention to the space - lol

**Comment by mizunos** [comment]
> Thanks a lot for addressing my feedback 🙏 I left some final nits and then we're good to me. Please fix the DCO commit signing (see https://cortexmetrics.io/docs/contributing/#developer-certificates-of-origin-dcos), otherwise we'll not be able to merge it.

Does it mean that I have to open a new pull request or only the last commit matter?

**Comment by mizunos** [comment]
@pracucci use commit 55561de for merging if all looks good.

File: docs/guides/alert-manager-configuration.md
Comment (pstibrany) [inline docs/guides/alert-manager-configuration.md:67]: missing `\` at the end of line

File: docs/guides/alert-manager-configuration.md
Comment (mizunos) [reply docs/guides/alert-manager-configuration.md:55]: Done

File: docs/guides/alert-manager-configuration.md
Comment (mizunos) [reply docs/guides/alert-manager-configuration.md:67]: Fixed

File: docs/guides/alert-manager-configuration.md
Comment (mizunos) [reply docs/guides/alert-manager-configuration.md:48]: ok.  I gave it the ID 100 instead

**Comment by mizunos** [comment]
Fix all DCO - Anything else here?.  still waiting for one more approval.

**Comment by pstibrany** [comment]
Thank you. Linter complains about this now:
```
docs/guides/alert-manager-configuration.md:62:51: "sucessfully" is a misspelling of "successfully"
6
docs/guides/alert-manager-configuration.md:66:2: "adress" is a misspelling of "address"
```

**Comment by pracucci** [comment]
>  Linter complains about this now:

I just pushed a commit to fix it.

**Review by pracucci — APPROVED** [review]
Thanks!

**Pull request merged**
